src/Bicep.Core/Registry/Oci/OciArtifactReferenceFacts.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Bicep.Core.Registry.Oci
+{
+    public static partial class OciArtifactReferenceFacts
+    {
+        public const string Scheme = "br";
+
+        public const int MaxRegistryLength = 255;
+
+        // must be kept in sync with the tag name regex
+        public const int MaxTagLength = 128;
+
+        public const int MaxRepositoryLength = 255;
+
+        // the registry component is equivalent to a host in a URI, which are case-insensitive
+        public static readonly IEqualityComparer<string> RegistryComparer = StringComparer.OrdinalIgnoreCase;
+
+        // repository component is case-sensitive (although regex blocks upper case)
+        public static readonly IEqualityComparer<string> RepositoryComparer = StringComparer.Ordinal;
+
+        // tags are case-sensitive and may contain upper and lowercase characters
+        public static readonly IEqualityComparer<string?> TagComparer = StringComparer.Ordinal;
+
+        // digests are case sensitive
+        public static readonly IEqualityComparer<string?> DigestComparer = StringComparer.Ordinal;
+
+        // must be kept in sync with the tag max length
+        private static readonly Regex TagRegex = new(@$"^[a-zA-Z0-9_][a-zA-Z0-9._-]{{0,{MaxTagLength - 1}}}$", RegexOptions.Compiled | RegexOptions.ExplicitCapture | RegexOptions.CultureInvariant);
```

Comment:
This can be converted to generated as well.

Reply:
will address in a separate PR